controllers/resources/manager.go

```diff
@@ -116,7 +116,15 @@ func (m *manager) GenerateRemediationCRForMachine(machine *machinev1beta1.Machin
 		// So it can be ignored here.
 	}
 
-	return m.generateRemediationCR(machine.GetName(), mhcOwnerRef, machineOwnerRef, template)
+	cr, err := m.generateRemediationCR(machine.GetName(), mhcOwnerRef, machineOwnerRef, annotations.RemoveMultipleTemplatesAnnotation(template))
+	//MHC does not support multiple templates so the annotations indicating the template aren't required
+	if err == nil && cr != nil {
+		ann := cr.GetAnnotations()
```

Comment:
Same here: adding the annotation first, and then removing it here, is unmaintainable. The logic in  `generateRemediationCR` should be changed to do the right thing. 

Also: as stated in earlier discussions, I'm afraid that not setting the annotation can cause side effects, because we use it at many code places. I would strongly prefer to set it to the right value (node name) instead.

Reply:
The rational behind keeping the changes in `GenerateRemediationCRForMachine` vs `generateRemediationCR` was to make sure the changes will only affect MHC created remediations in order to minimize further regression risks.

The alternatives IMO are either 
- adding a flag to generateRemediationCR in order to indicate MHC/NHC CR 
- or add the differentiation logic inside of generateRemediationCR (probably by checking the healthCheckOwnerRef) 
 
both of these are fine by me, so let me know if you have a specific strong preference, or had something else in mind

Reply:
healthCheckOwnerRef sounds like the way to go, no need for a flag

Reply:
looks like in any case `generateRemediationCR` needs to receive the nodeName (to set in the annotation in case of MHC remediation) so that probably will cover all of the bases.